src/Stratis.Bitcoin.Tests/P2P/PeerConnectorTests.cs

```diff
@@ -210,6 +210,136 @@ public void PeerConnectorDiscovery_ConnectsTo_DiscoveredPeers()
+            bool connectedToThisPeer = ConnectToPeer(peerAddressManager, networkPeerFactoryExisting, connectionManagerSettingsExisting, peerConnector, originalLocalhostNodeEndpoint);
+            Assert.True(connectedToThisPeer);
+
+            // Peer 2 on localhost on different port.
+            IPAddress addressInSameNetworkGroup = IPAddress.Parse("::ffff:127.0.0.1");
+            var secondLocalhostNodeEndpoint = new IPEndPoint(addressInSameNetworkGroup, 90);
+            connectedToThisPeer = ConnectToPeer(peerAddressManager, networkPeerFactoryExisting, connectionManagerSettingsExisting, peerConnector, secondLocalhostNodeEndpoint);
+            Assert.True(connectedToThisPeer);
+        }
+
+        [Fact]
+        public void PeerConnectorDiscovery_DoesNotConnectTo_NodeInSameNetworkGroup_WithIpRangeFilteringEnabled()
+        {
+            // Setup.
+            Network mainNetwork = Network.Main;
+            DataFolder peerFolder = CreateDataFolder(this);
+            var peerAddressManager = new PeerAddressManager(DateTimeProvider.Default, peerFolder, this.extendedLoggerFactory, new SelfEndpointTracker());
+            var nodeSettings = new NodeSettings(mainNetwork, args: new[] { "-IpRangeFiltering" });
+
+            var connectionManagerSettingsExisting = new ConnectionManagerSettings(nodeSettings);
+            Mock<INetworkPeerFactory> networkPeerFactoryExisting = new Mock<INetworkPeerFactory>();
+
+            var peerConnector = new PeerConnectorConnectNode(this.asyncLoopFactory, DateTimeProvider.Default, this.extendedLoggerFactory, this.network, networkPeerFactoryExisting.Object, this.nodeLifetime, nodeSettings, connectionManagerSettingsExisting, peerAddressManager, new SelfEndpointTracker());
+
+            IConnectionManager connectionManagerExisting = this.CreateConnectionManager(nodeSettings, connectionManagerSettingsExisting, peerAddressManager, peerConnector);
+            peerConnector.Initialize(connectionManagerExisting);
+
+            // Original peer.
+            IPAddress originalAddress = IPAddress.Parse("::ffff:192.168.0.1");
+            var originalNodeEndpoint = new IPEndPoint(originalAddress, 80);
+            bool connectToThisPeer = ConnectToPeer(peerAddressManager, networkPeerFactoryExisting, connectionManagerSettingsExisting, peerConnector, originalNodeEndpoint);
+            Assert.True(connectToThisPeer);
+            
+            // Peer from same group.
+            IPAddress addressFromNodeInSameGroup = IPAddress.Parse("::ffff:192.168.0.2");
+            var endpointOfNodeInSameGroup = new IPEndPoint(addressFromNodeInSameGroup, 80);
+            connectToThisPeer = ConnectToPeer(peerAddressManager, networkPeerFactoryExisting, connectionManagerSettingsExisting, peerConnector, endpointOfNodeInSameGroup);
+            Assert.False(connectToThisPeer);
+
+            // Peer from different group.
+            IPAddress addressFromNodeInDifferentGroup = IPAddress.Parse("::ffff:192.168.0.1");
```

Comment:
I'm confused about this.  We need to use non local addresses to test that part of the functionality, surely?

Reply:
I would expect some tests to work with normal IP version 6 addresses, not only IP version 4 addresses in IP version 6 format. You are only working with IP version 4 addresses everywhere. Moreover, mostly you use local network addresses, so it feels like the test cases are very limited. There are different rules for IP version 6 addresses so if you just use IP version 4 addresses, even if they are converted to IP version 6 format, you will certainly not test how does it behave for a IP version six addresses. Similarly, if you mostly use local network addresses, it will not test nonlocal network addresses scenarios.

Reply:
Have changed to use a mix of ipv4 and ipv6.  If these need changing or extending pls comment